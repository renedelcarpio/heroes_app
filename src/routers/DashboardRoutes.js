import { Route, Routes } from 'react-router-dom';

import DcScreen from '../components/dc/DcScreen';
import HeroScreen from '../components/hero/HeroScreen';
import MarvelScreen from '../components/marvel/MarvelScreen';
import { Navbar } from '../components/ui/Navbar';
import SearchScreen from '../components/search/SearchScreen';

const DashboardRoutes = () => {
	return (
		<>
			<Navbar />
			<Routes>
				<Route path='marvel' element={<MarvelScreen />} />
				<Route path='dc' element={<DcScreen />} />
				<Route path='search' element={<SearchScreen />} />
				<Route path='hero/:heroId' element={<HeroScreen />} />
				<Route path='/' element={<MarvelScreen />} />
			</Routes>
		</>
	);
};

export default DashboardRoutes;
