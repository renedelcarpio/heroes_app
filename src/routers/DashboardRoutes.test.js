import { AuthContext } from '../auth/authContext';
import DashboardRoutes from './DashboardRoutes';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

describe('Tests in DashboardRoutes component', () => {
	const contextValue = {
		user: {
			logged: true,
			name: 'userName',
		},
	};

	test('should render correctly Marvel', () => {
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<MemoryRouter initialEntries={['/']}>
					<DashboardRoutes />
				</MemoryRouter>
			</AuthContext.Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.user-name').text().trim()).toBe('userName');
		expect(wrapper.find('h1').text().trim()).toBe('Marvel Super Heroes');
	});

	test('should render correctly Dc', () => {
		const wrapper = mount(
			<AuthContext.Provider value={contextValue}>
				<MemoryRouter initialEntries={['/dc']}>
					<DashboardRoutes />
				</MemoryRouter>
			</AuthContext.Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('h1').text().trim()).toBe('DC Super Heroes');
	});
});
