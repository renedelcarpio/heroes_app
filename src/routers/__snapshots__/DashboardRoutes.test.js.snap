// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in DashboardRoutes component should render correctly Dc 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="navbar__brand"
      href="/"
      onClick={[Function]}
    >
      Heroes App
    </a>
    <div
      className="navbar__heroes"
    >
      <div
        className="navbar__nav"
      >
        <a
          className="navbar__nav-item "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="navbar__nav-item dc-active"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="navbar__nav-item "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar__login"
    >
      <ul
        className="navbar__nav name-button"
      >
        <span
          className="navbar__nav-item user-name"
        >
          userName
        </span>
        <button
          className="navbar__nav-item button"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="dc__container"
  >
    <div
      className="dc__image-container"
    >
      <img
        alt="DC Logo"
        className="dc__image"
        src="https://wallpaper.dog/large/17219721.jpg"
      />
    </div>
    <h1
      className="dc__title"
    >
      DC Super Heroes
    </h1>
    <div
      className="dc__cards-container"
    >
      <div
        className="card__container"
      >
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Batman"
              height={300}
              src="/assets/dc-batman.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Batman
            </h2>
            <h3>
              Bruce Wayne
            </h3>
            <a
              className="card__link"
              href="/hero/dc-batman"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Superman"
              height={300}
              src="/assets/dc-superman.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Superman
            </h2>
            <h3>
              Kal-El
            </h3>
            <a
              className="card__link"
              href="/hero/dc-superman"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Flash"
              height={300}
              src="/assets/dc-flash.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Flash
            </h2>
            <h3>
              Jay Garrick
            </h3>
            <p>
              Jay Garrick, Barry Allen, Wally West, Bart Allen
            </p>
            <a
              className="card__link"
              href="/hero/dc-flash"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Green Lantern"
              height={300}
              src="/assets/dc-green.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Green Lantern
            </h2>
            <h3>
              Alan Scott
            </h3>
            <p>
              Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
            </p>
            <a
              className="card__link"
              href="/hero/dc-green"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Green Arrow"
              height={300}
              src="/assets/dc-arrow.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Green Arrow
            </h2>
            <h3>
              Oliver Queen
            </h3>
            <a
              className="card__link"
              href="/hero/dc-arrow"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Wonder Woman"
              height={300}
              src="/assets/dc-wonder.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Wonder Woman
            </h2>
            <h3>
              Princess Diana
            </h3>
            <a
              className="card__link"
              href="/hero/dc-wonder"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Martian Manhunter"
              height={300}
              src="/assets/dc-martian.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Martian Manhunter
            </h2>
            <h3>
              J'onn J'onzz
            </h3>
            <p>
              Martian Manhunter
            </p>
            <a
              className="card__link"
              href="/hero/dc-martian"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Robin/Nightwing"
              height={300}
              src="/assets/dc-robin.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Robin/Nightwing
            </h2>
            <h3>
              Dick Grayson
            </h3>
            <a
              className="card__link"
              href="/hero/dc-robin"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Blue Beetle"
              height={300}
              src="/assets/dc-blue.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Blue Beetle
            </h2>
            <h3>
              Dan Garret
            </h3>
            <p>
              Dan Garret, Ted Kord, Jaime Reyes
            </p>
            <a
              className="card__link"
              href="/hero/dc-blue"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Black Canary"
              height={300}
              src="/assets/dc-black.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Black Canary
            </h2>
            <h3>
              Dinah Drake
            </h3>
            <p>
              Dinah Drake, Dinah Lance
            </p>
            <a
              className="card__link"
              href="/hero/dc-black"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Tests in DashboardRoutes component should render correctly Marvel 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      className="navbar__brand"
      href="/"
      onClick={[Function]}
    >
      Heroes App
    </a>
    <div
      className="navbar__heroes"
    >
      <div
        className="navbar__nav"
      >
        <a
          className="navbar__nav-item "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="navbar__nav-item "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="navbar__nav-item "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar__login"
    >
      <ul
        className="navbar__nav name-button"
      >
        <span
          className="navbar__nav-item user-name"
        >
          userName
        </span>
        <button
          className="navbar__nav-item button"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="marvel__container"
  >
    <div
      className="marvel__image-container"
    >
      <img
        alt="DC Logo"
        className="marvel__image"
        src="https://thecouch.world/wp-content/uploads/2018/02/Marvel-Comics-Logo-Desktop-Wallpaperconv.jpg"
      />
    </div>
    <h1
      className="marvel__title"
    >
      Marvel Super Heroes
    </h1>
    <div
      className="marvel__cards-container"
    >
      <div
        className="card__container"
      >
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Spider Man"
              height={300}
              src="/assets/marvel-spider.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Spider Man
            </h2>
            <h3>
              Peter Parker
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-spider"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Captain America"
              height={300}
              src="/assets/marvel-captain.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Captain America
            </h2>
            <h3>
              Steve Rogers
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-captain"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Iron Man"
              height={300}
              src="/assets/marvel-iron.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Iron Man
            </h2>
            <h3>
              Tony Stark
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-iron"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Thor"
              height={300}
              src="/assets/marvel-thor.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Thor
            </h2>
            <h3>
              Thor Odinson
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-thor"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Hulk"
              height={300}
              src="/assets/marvel-hulk.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Hulk
            </h2>
            <h3>
              Bruce Banner
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-hulk"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Wolverine"
              height={300}
              src="/assets/marvel-wolverine.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Wolverine
            </h2>
            <h3>
              James Howlett
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-wolverine"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Daredevil"
              height={300}
              src="/assets/marvel-daredevil.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Daredevil
            </h2>
            <h3>
              Matthew Michael Murdock
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-daredevil"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Hawkeye"
              height={300}
              src="/assets/marvel-hawkeye.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Hawkeye
            </h2>
            <h3>
              Clinton Francis Barton
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-hawkeye"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Cyclops"
              height={300}
              src="/assets/marvel-cyclops.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Cyclops
            </h2>
            <h3>
              Scott Summers
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-cyclops"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card__image-container"
          >
            <img
              alt="Silver Surfer"
              height={300}
              src="/assets/marvel-silver.jpg"
              width={250}
            />
          </div>
          <div
            className="card__info-container"
          >
            <h2>
              Silver Surfer
            </h2>
            <h3>
              Norrin Radd
            </h3>
            <a
              className="card__link"
              href="/hero/marvel-silver"
              onClick={[Function]}
            >
              More...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
